packages/runtime-core/src/vnode.ts

```diff
@@ -103,12 +103,50 @@ export type VNodeProps = {
   ref_for?: boolean
   ref_key?: string
 
+  // vue hooks
+  ['onVue:BeforeMount']?: VNodeMountHook | VNodeMountHook[]
+  ['onVue:Mounted']?: VNodeMountHook | VNodeMountHook[]
+  ['onVue:BeforeUpdate']?: VNodeUpdateHook | VNodeUpdateHook[]
+  ['onVue:Updated']?: VNodeUpdateHook | VNodeUpdateHook[]
+  ['onVue:BeforeUnmount']?: VNodeMountHook | VNodeMountHook[]
+  ['onVue:Unmounted']?: VNodeMountHook | VNodeMountHook[]
```

Comment:
Sorry if this is a dumb question, TypeScript isn't my strong suit, but shouldn't these follow Vue's usual convention of having a lowercase letter directly after the colon? e.g. `onVue:beforeMount`.

Reply:
My mistake here, it should be lowercase, I was sure I lowered case them :D